test/extensions/filters/listener/tls_inspector/tls_inspector_benchmark.cc

```diff
@@ -93,18 +77,21 @@ static void BM_TlsInspector(benchmark::State& state) {
   TestThreadsafeSingletonInjector<Api::OsSysCallsImpl> os_calls{&os_sys_calls};
   NiceMock<Stats::MockStore> store;
   ConfigSharedPtr cfg(std::make_shared<Config>(store));
-  FastMockConnectionSocket socket;
+  Network::ConnectionSocketImpl socket(42, nullptr, nullptr);
```

Comment:
This is going to result in socket calls on fd 42. I think you should call socket() to get a real fd to avoid flakiness by making random syscalls on a possibly-used fd.

Reply:
Good catch, thanks!

I did some tracing, and the only syscall that was happening was `close(42)`, which didn't affect results, and is now avoided by using invalid `fd`... All other syscalls are mocked via `Api::MockOsSysCalls`, so it didn't matter for them (`close()` is called directly and not via `Api::OsSysCallsImpl` in a few places - I'm going open separate PR for that).